[Filestream] Expand take-over mode for Filestream inputs and support dynamic inputs (#42624)
elastic/beats: docs/reference/filebeat/_step_2_enable_the_take_over_mode.md

# Step 2: Enable the take over mode [_step_2_enable_the_take_over_mode]

Now, to indicate that the new `filestream` is supposed to take over
the files from a previously defined `log` input, we need to add
`take_over.enabled: true` to each new `filestream`. This will make
sure that the new `filestream` inputs will continue ingesting files
from the same offset where the `log` inputs stopped.

::::{note}
After the first run with `take_over.enabled: true` the setting can be removed.
::::

::::{warning}
The `take over` mode is in beta.
::::

::::{important}

> Only files `filestream` find during its initialisation will have the
> state migrated.

Reviewer: I'm not sure this would be meaningful for the users. If you decide to keep it, I'd rephrase it.

```suggestion
Only states of still existing files will be migrated during the `filestream` initialization.
```

Author: I believe it is useful, I want to make clear the limitations.

Log files that have been ingested don't disappear and then reappear in a normal workflow, but I wanted to make clear that if the file has been removed (the path does not match any of the globs defined in the input), then its state will not be migrated.
